### PR TITLE
Use TRAVIS_BRANCH to select branch of toolchain being tested

### LINKED ISSUE #125: Ensure that master branch is testing the correct branches of the toolchain
In #124, we use the option `--branch` in `git clone` for all tests and provide the default value `develop` (this approach is consistent with the preCICE baseimages). We also changed the default branch of many of our repositories to `develop`. Now we have to make sure that `systemtests:master` tests the `master` of all repositories (**not** `develop`, which is the default).

1. ~~A proper solution would be to implement the functionality suggested in #121 and use this to tell the test on `systemtests:master` to use the `master` branch of all repositories - ideally via the `.travis.yaml` or another even smarter CI feature.~~ (not necessary for closing this issue. We go for option 2)

2. The less proper but quick solution would be to use the already existing `--branch` argument of `system_testing.py` to use the provided branch not only for the creation of the `precice` image, but for all images. (see #136)

### PR DESCRIPTION
**Pull request by BenjaminRodenberg**
closes #125 

# Todo

- [x] depends on https://github.com/precice/systemtests/pull/174. Has to be merged first.

### CODE REVIEW

**Comment by BenjaminRodenberg** [comment]
https://travis-ci.org/precice/systemtests/jobs/623725505?utm_medium=notification&utm_source=github_status shows that we need some more logic in this line of `.travis.yml`:

https://github.com/precice/systemtests/blob/1cc255277f54817442ed6a53022a105fc5560bbe/.travis.yml#L18

It will probably be sufficient to use something similar to the approach used in `system_testing.py`:

https://github.com/precice/systemtests/blob/751a263b30eb30b329d4f23acac33ab3a9460729/system_testing.py#L178

As soon as #131 is merged, we can directly reuse this logic in the python scripts for building base images that are introduced in #131. 

Therefore, let's first merge #131, then continue here.

File: .travis.yml [hunk 1]
```diff
@@ -139,41 +139,41 @@ jobs:
       name: SU2 adapter
       if: fork = false
       script:
-        - docker build -f adapters/Dockerfile.su2-adapter -t $DOCKER_USERNAME/su2-adapter-ubuntu1604.home-develop .
+        - docker build -f adapters/Dockerfile.su2-adapter -t $DOCKER_USERNAME/su2-adapter-ubuntu1604.home-$TRAVIS_BRANCH --build-arg branch=$TRAVIS_BRANCH .
```
Comment (BenjaminRodenberg) [inline .travis.yml]:
```suggestion
        - docker build -f adapters/Dockerfile.su2-adapter -t $DOCKER_USERNAME/su2-adapter-ubuntu1604.home-$TRAVIS_BRANCH --build-arg branch=$TRAVIS_BRANCH --build-arg from=$DOCKER_USERNAME/precice-ubuntu1604.home-$TRAVIS_BRANCH .
```

File: .travis.yml [hunk 2]
```diff
@@ -139,41 +139,41 @@ jobs:
...
         script: >-
           echo "$DOCKER_PASSWORD" | docker login -u $DOCKER_USERNAME --password-stdin &&
-          docker push $DOCKER_USERNAME/su2-adapter-ubuntu1604.home-develop
+          docker push $DOCKER_USERNAME/su2-adapter-ubuntu1604.home-$TRAVIS_BRANCH
 
     - name: CalculiX adapter
       if: fork = false
       script:
-        - docker build -f adapters/Dockerfile.calculix-adapter -t $DOCKER_USERNAME/calculix-adapter-ubuntu1604.home-develop .
+        - docker build -f adapters/Dockerfile.calculix-adapter -t $DOCKER_USERNAME/calculix-adapter-ubuntu1604.home-$TRAVIS_BRANCH --build-arg branch=$TRAVIS_BRANCH .
```
Comment (BenjaminRodenberg) [inline .travis.yml]:
```suggestion
        - docker build -f adapters/Dockerfile.calculix-adapter -t $DOCKER_USERNAME/calculix-adapter-ubuntu1604.home-$TRAVIS_BRANCH --build-arg branch=$TRAVIS_BRANCH --build-arg from=$DOCKER_USERNAME/precice-ubuntu1604.home-$TRAVIS_BRANCH .
```

File: .travis.yml [hunk 3]
```diff
@@ -139,41 +139,41 @@ jobs:
...
       deploy:
         skip_cleanup: true
         provider: script
         on:
           all_branches: true
         script: >-
-          echo "$DOCKER_PASSWORD" | docker login -u $DOCKER_USERNAME --password-stdin &&
-          docker push $DOCKER_USERNAME/calculix-adapter-ubuntu1604.home-develop
+          echo "$DOCKER_PASSWORD" | docker login -u "precice" --password-stdin &&
```
Comment (BenjaminRodenberg) [inline .travis.yml]:
```suggestion
          echo "$DOCKER_PASSWORD" | docker login -u $DOCKER_USERNAME --password-stdin &&
```

File: .travis.yml [hunk 4]
```diff
@@ -139,41 +139,41 @@ jobs:
...
+          docker push precice/calculix-adapter-ubuntu1604.home-$TRAVIS_BRANCH
```
Comment (BenjaminRodenberg) [inline .travis.yml]:
```suggestion
          docker push $DOCKER_USERNAME/calculix-adapter-ubuntu1604.home-$TRAVIS_BRANCH
```

File: .travis.yml [hunk 5]
```diff
@@ -139,41 +139,41 @@ jobs:
...
 
     - name: deal.ii adapter
       if: fork = false
       script:
-        - docker build -f adapters/Dockerfile.dealii-adapter -t $DOCKER_USERNAME/dealii-adapter-ubuntu1604.home-develop .
+        - docker build -f adapters/Dockerfile.dealii-adapter -t $DOCKER_USERNAME/dealii-adapter-ubuntu1604.home-$TRAVIS_BRANCH --build-arg branch=$TRAVIS_BRANCH .
```
Comment (BenjaminRodenberg) [inline .travis.yml]:
```suggestion
        - docker build -f adapters/Dockerfile.dealii-adapter -t $DOCKER_USERNAME/dealii-adapter-ubuntu1604.home-$TRAVIS_BRANCH --build-arg branch=$TRAVIS_BRANCH --build-arg from=$DOCKER_USERNAME/precice-ubuntu1604.home-$TRAVIS_BRANCH .
```

File: .travis.yml [hunk 6]
```diff
@@ -198,53 +198,53 @@ jobs:
         on:
           all_branches: true
         script: >-
-          echo "$DOCKER_PASSWORD" | docker login -u "precice" --password-stdin &&
+          echo "$DOCKER_PASSWORD" | docker login -u $DOCKER_USERNAME --password-stdin &&
           docker push precice/openfoam-adapter-ubuntu1804.home-develop
```
Comment (BenjaminRodenberg) [inline .travis.yml]:
```suggestion
          docker push $DOCKER_USERNAME/openfoam-adapter-ubuntu1804.home-$TRAVIS_BRANCH
```

File: .travis.yml [hunk 7]
```diff
@@ -198,53 +198,53 @@ jobs:
...
+          docker push $DOCKER_USERNAME/openfoam-adapter-ubuntu1604.home.petsc-$TRAVIS_BRANCH
 
     - name: CalculiX adapter [PETSc] [Job failure permitted]
       if: fork = false
       script:
         - >
-          docker build -f adapters/Dockerfile.calculix-adapter -t $DOCKER_USERNAME/calculix-adapter-ubuntu1604.home.petsc-develop
-          --build-arg from=$DOCKER_USERNAME/precice-ubuntu1604.home.petsc-develop .
+          docker build -f adapters/Dockerfile.calculix-adapter -t $DOCKER_USERNAME/calculix-adapter-ubuntu1604.home.petsc-$TRAVIS_BRANCH
+          --build-arg from=precice/precice-ubuntu1604.home.petsc-$TRAVIS_BRANCH --build-arg branch=$TRAVIS_BRANCH .
```
Comment (BenjaminRodenberg) [inline .travis.yml]:
```suggestion
          --build-arg from=$DOCKER_USERNAME/precice-ubuntu1604.home.petsc-$TRAVIS_BRANCH --build-arg branch=$TRAVIS_BRANCH .
```

File: .travis.yml [hunk 8]
```diff
@@ -198,53 +198,53 @@ jobs:
...
       deploy:
         skip_cleanup: true
         provider: script
         on:
           all_branches: true
         script: >-
           echo "$DOCKER_PASSWORD" | docker login -u "$DOCKER_USERNAME" --password-stdin &&
```
Comment (BenjaminRodenberg) [inline .travis.yml]:
```suggestion
          echo "$DOCKER_PASSWORD" | docker login -u $DOCKER_USERNAME --password-stdin &&
```

**Review by BenjaminRodenberg — COMMENTED** [review]
We have to consequently use `--build-arg` `from` and `branch`

**Comment by BenjaminRodenberg** [comment]
I'm checking the correct behaviour for `develop/master` for this feature on my fork. See https://travis-ci.com/BenjaminRueth/systemtests/branches?utm_medium=notification&utm_source=email

**Comment by BenjaminRodenberg** [comment]
> Therefore, let's first merge #131, then continue here.

Without #131, this PR is a bit more complicated, but working. Therefore there is no necessary dependency between #131 and #136. Let's try to work on them independently.

**Review by Eder-K — APPROVED** [review]

**Comment by BenjaminRodenberg** [comment]
As soon as #173 is merged, I would suggest to continue here.

**Comment by BenjaminRodenberg** [comment]
I merged `develop` into this branch. I.e. additions added recently via #131 are now merged into this PR. I will again check the changes on my fork: https://travis-ci.com/BenjaminRueth/systemtests/branches?utm_medium=notification&utm_source=email

**Comment by BenjaminRodenberg** [comment]
Eigen seems to currently face an issue: https://travis-ci.com/BenjaminRueth/systemtests/jobs/290050258#L1437-L1438

This problem currently also occurs, if I try to open https://gitlab.com/libeigen/eigen/-/archive/3.3.7/eigen-3.3.7.tar.bz2 in my browser. Let's hope that waiting helps:

Screenshot from 2020-02-22 13-43-27

**Comment by BenjaminRodenberg** [comment]
The Eigen Problem seems to be resolved.

**Pull request merged**